frontend/src/plugin/loadPlugins.js

```diff
@@ -0,0 +1,20 @@
+export default function loadScripts(array, callback){
+  var loader = function(src, handler){
+    var script = document.createElement('script');
+    script.src = src;
+    script.onload = script.onreadystatechange = function(){
+      script.onreadystatechange = script.onload = null;
+      handler();
+    };
+    var head = document.getElementsByTagName('head')[0];
+    (head || document.body).appendChild( script );
+  };
+  (function run(){
+    if (array.length !== 0){
+      loader(array.shift(), run);
+    } else {
+      callback && callback();
+    }
+  })();
```

Comment:
I find this code a bit difficult to follow. Maybe you can explain this better or point to where you got this if you copied bits.

Reply:
So we want to fetch all the pluginScripts and the above function job is to be called whenever a previous script in the array is loaded, with the updated array(by deleting the first element using array.shift and now for the loadScript the updated first element is in focus), and we do it until our array has become empty(in case of empty we do our work which we want to do when all script are loaded i.e., in our case initialize plugins).
I think the IIFE part is confusing here It is in no way different from calling the run function manually in the next line.